app/src/main/java/ch/epfl/sdp/peakar/user/services/providers/firebase/FirebaseAuthService.java

```diff
@@ -0,0 +1,160 @@
+package ch.epfl.sdp.peakar.user.services.providers.firebase;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.firebase.ui.auth.AuthUI;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FacebookAuthProvider;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.GoogleAuthProvider;
+
+import java.util.Objects;
+
+import ch.epfl.sdp.peakar.general.remote.RemoteOutcome;
+import ch.epfl.sdp.peakar.general.remote.RemoteResource;
+import ch.epfl.sdp.peakar.user.services.Account;
+import ch.epfl.sdp.peakar.user.services.AuthProvider;
+import ch.epfl.sdp.peakar.user.services.AuthService;
+
+/**
+ * This class describes the Auth service provided by Firebase.
+ * It makes use of a FirebaseAccount as this authentication service is always used only if Firebase Realtime Database is available.
+ */
+public class FirebaseAuthService implements AuthService {
+    private static FirebaseAuthService instance;
+
+    // The account reference will be null if no account is authenticated, or != null if an account is authenticated
+    private static FirebaseAccount authAccount;
+
+    private FirebaseAuthService() {}
+
+    public static FirebaseAuthService getInstance() {
+        if(instance == null) {
+            instance = new FirebaseAuthService();
+            // On class initialization, retrieve any previously logged account and, if necessary, the account data
+            authAccount = FirebaseAuth.getInstance().getCurrentUser() != null ? FirebaseAccount.getInstance(FirebaseAuth.getInstance().getCurrentUser().getUid()) : null;
+            if(authAccount != null) new Thread (() -> authAccount.retrieveData()).start();
+        }
+        return instance;
+    }
+
+    @Override
+    public RemoteOutcome authWithProvider(AuthProvider authProvider, String token) {
+
+        RemoteOutcome outcome = RemoteOutcome.FAIL;
```

Comment:
Initialization of outcome is redundant.
```suggestion
        RemoteOutcome outcome;
```

Reply:
Removed initialization.